### ISSUE TITLE
IE 11 issues

### ISSUE THREAD

**Issue by p1ho** [oscarmorrison/md-page#45]
It would appear that IE may require more polyfills than we initially thought.
![image](https://user-images.githubusercontent.com/22629067/81779230-276c1500-94a9-11ea-8e33-d0cde4b4ae4e.png)
I can try to get to the bottom of this.

Relevant:
#43 #44

**Comment by p1ho** [comment]
Still doesn't work 100%... this may need a deeper dive.
![image](https://user-images.githubusercontent.com/22629067/81781925-d579be00-94ad-11ea-8106-5d87914a9ce0.png)

**Comment by p1ho** [comment]
I have tried updating  [`./showdown.js`](https://github.com/oscarmorrison/md-page/blob/review/src/showdown.js) from the [showdown repo](https://github.com/showdownjs/showdown/blob/master/dist/showdown.js), but still doesn't change anything.
I recommend testing out showdown's compatability with IE11 first, then we can go from there.

**Comment by oscarmorrison** [comment]
Yep sounds good. IE is a terrible browser. I don’t like accounting for it. So if people want a fix happy to accept PR. But not keen to do myself

**Comment by oscarmorrison** [comment]
> Still doesn't work 100%... but no errors. This may need a deeper dive.
> ![image](https://user-images.githubusercontent.com/22629067/81781925-d579be00-94ad-11ea-8106-5d87914a9ce0.png)

Can you do a log of the showdown js

**Comment by p1ho** [comment]
sorry, what kind of log are you looking for?

**Comment by oscarmorrison** [comment]
Like the console output of this: https://github.com/oscarmorrison/md-page/blob/master/src/script.js#L86

To work out whether IE11 is being silly and not rendering the blocks, or because showdown is not working

**Comment by p1ho** [comment]
Yeah there's definitely something wrong with the way IE11 processes showdown.
here's the test html
```html
<script src="https://livejs.com/live.js"></script>
<script src="src/showdown.js"></script>
<script src="src/script.js"></script><noscript>

# test h1
## test h2
### test h3
#### test h4
##### test h5
###### test h6
```

spits out 
```
<p># test h1 ## test h2 ### test h3 #### test h4 ##### test h5 ###### test h6</p>
```
in IE 11,

whereas the same file spits out the correct thing in Chrome/Firefox
```
<h1 id="test-h1">test h1</h1>
<h2 id="test-h2">test h2</h2>
<h3 id="test-h3">test h3</h3>
<h4 id="test-h4">test h4</h4>
<h5 id="test-h5">test h5</h5>
<h6 id="test-h6">test h6</h6>
```

**Comment by oscarmorrison** [comment]
Yep. Thanks @p1ho wanted to work out what was exact error.
Next steps would be to verify showdown normally works with IE11

can you try: http://demo.showdownjs.com/

If that works, then we should remove showdown manually dep, and use npm for it

**Comment by p1ho** [comment]
That works, yeah I'll try including it via npm, see if that works

**Comment by oscarmorrison** [comment]
sounds good, thankyou

**Comment by p1ho** [comment]
It doesn't work, it's still outputting the wrong string.
```html
<script src="https://livejs.com/live.js"></script>
<script src="node_modules/showdown/dist/showdown.js"></script>
<script src="src/script.js"></script><noscript>
```

**Comment by oscarmorrison** [comment]
what about with:   
https://cdnjs.cloudflare.com/ajax/libs/showdown/1.9.1/showdown.min.js

**Comment by oscarmorrison** [comment]
then running in console 
```
var converter = new showdown.Converter(),
    text      = '# hello, markdown!',
    html      = converter.makeHtml(text);
```

**Comment by p1ho** [comment]
^ That works!

**Comment by oscarmorrison** [comment]
So what about just 
```
<script src="https://cdnjs.cloudflare.com/ajax/libs/showdown/1.9.1/showdown.min.js"></script>
<script src="src/script.js"></script><noscript>
```

**Comment by p1ho** [comment]
Found the problem, this line
```
    var markdown = document.querySelector('noscript').innerText
```
in IE 11 returns a string without new line

**Comment by p1ho** [comment]
Alright, I got it to work!

Basically, [IE 11 will only preserve text formatting **on text extraction** for `<pre>` and `<textarea>`](https://stackoverflow.com/questions/656605/jquery-text-call-preserves-newlines-in-firefox-but-not-in-ie), we could try to use any of the 2 instead of `<noscript>`, but this breaks backward compatibility (people who still had `<noscript>` in their code will find that it stopped working).

So the trick is, when browser is IE 11, we clone the `<noscript>` text content into a `<pre>` (note: this preserves formatting because it's not **text extraction**), then extract text from the temp `<pre>` element.

![image](https://user-images.githubusercontent.com/22629067/81883435-16bfab80-954a-11ea-9f19-41b7e4d1a061.png)

Will open a PR

**Comment by oscarmorrison** [comment]
nice work. yeah that sounds good.
i think its very important to be able to retain:

```
<script type="text/javascript" src="https://rawcdn.githack.com/oscarmorrison/md-page/abe58a8f057636e8d865324b5f5821c760f3f6e9/md-page.js"></script><noscript>
```

**Comment by p1ho** [comment]
As long as people have been using the following
```
<!-- my-page.html --> 
<script src="https://rawcdn.githack.com/oscarmorrison/md-page/master/md-page.js"></script><noscript>
```
as instructed in the README.md, this shouldn't break anything for existing users.

**Comment by oscarmorrison** [comment]
can do it for ie only doing some like:

```
if (window.navigator.userAgent.indexOf("MSIE ")){
}
```

**Comment by p1ho** [comment]
@ji-ryoo You should be able to use it in IE 11 no problem now, feel free to open issues if you find something else not working. Thanks for letting us know.

**Comment by oscarmorrison** [comment]
Nice work @p1ho 

Closed in #47
